### PR TITLE
Making CCDData.read() less strict for input

### LINKED ISSUE #6386: WCSAXES not found error of CCDData
I have downloaded a random r-band image from SDSS DR12 ( [image search](https://dr12.sdss.org/fields/name?name=ngc+1132) and [download link](https://dr12.sdss.org/sas/dr12/boss/photoObj/frames/301/6572/6/frame-g-006572-6-0210.fits.bz2)) and saved as "ngc1132_r.fits".
I wanted to load this image but the ``CCDData`` gave ``KeyError``:

```python
from astropy.nddata import CCDData
from astropy.io import fits
from astropy.wcs import WCS

ccd = CCDData.read('data/ngc1132_r.fits')
```

The following is the error message:
```
WARNING: FITSFixedWarning: RADECSYS= 'ICRS ' / International Celestial Ref. System 
the RADECSYS keyword is deprecated, use RADESYSa. [astropy.wcs.wcs]
Traceback (most recent call last):

  File "<ipython-input-28-06a8bd5b201d>", line 1, in <module>
    ccd = CCDData.read('data/ngc1132_r.fits', unit='adu')

  File "/home/ysbach/anaconda3/lib/python3.6/site-packages/astropy/nddata/mixins/ndio.py", line 29, in read
    return io_registry.read(cls, *args, **kwargs)

  File "/home/ysbach/anaconda3/lib/python3.6/site-packages/astropy/io/registry.py", line 531, in read
    data = reader(*args, **kwargs)

  File "/home/ysbach/anaconda3/lib/python3.6/site-packages/astropy/nddata/ccddata.py", line 521, in fits_ccddata_reader
    hdr, wcs = _generate_wcs_and_update_header(hdr)

  File "/home/ysbach/anaconda3/lib/python3.6/site-packages/astropy/nddata/ccddata.py", line 419, in _generate_wcs_and_update_header
    del new_hdr[k]

  File "/home/ysbach/anaconda3/lib/python3.6/site-packages/astropy/io/fits/header.py", line 226, in __delitem__
    raise KeyError("Keyword '{}' not found.".format(key))

KeyError: "Keyword 'WCSAXES' not found."
```

The original file seems to have correct WCS and the following code works without any problem:

```python
from astropy.io import fits
from astropy.wcs import WCS

hdul = fits.open('data/ngc1132_r.fits')
w = WCS(hdul[0].header)
```

What does it mean by the ``KeyError`` of ``CCDData``? Should I just use ``astropy.io.fits`` than ``CCDData`` in this case?

### PR DESCRIPTION
**Pull request by bsipocz**
This should fix #6386 and another related issue with the too strict unit parsing.

### CODE REVIEW

**Comment by bsipocz** [comment]
Oh, @nden was quicker with #6389. Though the unit fix is still needed to be able to read the file refered to in #6386

**Comment by nden** [comment]
@bsipocz never mind, I'll close the other one.

**Comment by nden** [comment]
@bsipocz Go ahead with this one as it is more complete.

**Comment by bsipocz** [comment]
Don't close it, this was the duplicate. Could you though add the ``parse_strict='warn'`` change, there, too?

**Comment by nden** [comment]
I think I should move away from the keyboard, leaving this to you :-)

**Comment by bsipocz** [comment]
@nden - I took over your commit then and closed the other PR.

**Comment by mwcraig** [comment]
@bsipocz -- the unit behavior is deliberate 😀. We could change it, but would like @crawfordsm and @MSeifert04  to chime in ideally.

Would you be ok with backing that change out and we can open a separate PR/issue to make that change?

Tests are failing because unit errors are not being raised anymore.

**Comment by bsipocz** [comment]
@mwcraig - very well, it was a bit confusing bug report by claiming that reading the file mentioned in #6386 used to work with ccdproc. Apparently it didn't https://github.com/astropy/astropy/issues/6386#issuecomment-317257289

File: astropy/nddata/ccddata.py [hunk 1]
```diff
@@ -416,7 +416,11 @@ def _generate_wcs_and_update_header(hdr):
     wcs_header = wcs.to_header(relax=True)
     for k in wcs_header:
         if k not in _KEEP_THESE_KEYWORDS_IN_HEADER:
-            del new_hdr[k]
+            try:
+                del new_hdr[k]
+            except KeyError:
+                # Keyword does not exist in the original header
+                pass
```
Comment (saimn) [inline astropy/nddata/ccddata.py]: There is a [`Header.remove`](http://docs.astropy.org/en/latest/io/fits/api/headers.html#astropy.io.fits.Header.remove) method for this: `new_hdr.remove(k, ignore_missing=True)`

Comment (bsipocz) [inline astropy/nddata/ccddata.py]: brilliant, thanks! I'll update this shortly.

**Comment by bsipocz** [comment]
@mwcraig - very well, it was a bit confusing bug report by claiming that reading the file mentioned in #6386 used to work with ccdproc. Apparently it didn't https://github.com/astropy/astropy/issues/6386#issuecomment-317257289

File: astropy/nddata/ccddata.py (hunk 1 above)
Comment (MSeifert04) [inline astropy/nddata/ccddata.py]: I was going to suggest `Header.pop` with default but `Header.remove` is even better. :)

**Comment by crawfordsm** [comment]
I'd definitely separate out the two issues.   We originally decided to require units for a `CCDData` object, and I think that should stay or be discussed in more detail vs. fixing the bug in removing the WCS header information which should be a quicker and more minimal fix

**Comment by bsipocz** [comment]
@crawfordsm - Done, I've removed the unit related changes. As said above the initial bug report misled me of thinking it was working before in ccdproc rather than being a deliberate requirement.

**Comment by bsipocz** [comment]
@mwcraig @crawfordsm - I've addressed the issues, can we merge this for 2.0.1?

**Comment by bsipocz** [comment]
@mwcraig - Could you have a look at it, too?

**Comment by eteq** [comment]
@mwcraig or @crawfordsm - I merged this so that it makes it into 2.0.1, but feel free to leave comments if you had more to add!

@bsipocz, perhaps add a separate PR with just the unit change (presumably milestoned for 3.0), and that topic can be discussed there?

**Comment by crawfordsm** [comment]
Thanks @bsipocz and @eteq for merging

**Pull request merged**